graphql/src/execution/execution.rs

```diff
@@ -106,6 +154,20 @@ pub fn execute_root_selection_set(
     ctx: &ExecutionContext<impl Resolver>,
     selection_set: &q::SelectionSet,
 ) -> Result<BTreeMap<String, q::Value>, Vec<QueryExecutionError>> {
+    if CACHED_SUBGRAPH_IDS.contains(&ctx.query.schema.id) {
+        // Check if the response is cached.
+        let mut cache = QUERY_CACHE.lock().unwrap();
+
+        // Remove expired entries.
+        cache.notify_iter();
+
+        // Peek because we want even hot entries to invalidate after the expiry period.
+        if let Some(response) = cache.peek(&cache_key(ctx, selection_set)) {
+            debug!(ctx.logger, "Query cache hit");
```

Comment:
Consider removing this log. Logging in bulk can be a huge drain on performance. If this query is very hot this may be quite a lot of logs. A metric might be better.